lib/hooks.ts

```diff
@@ -75,3 +75,24 @@ export const useOrder = (orderId: number) => {
         error,
     };
 }
+
+export const useShippingAndProductsOrder = (orderId: number) => {
+    const { context } = useSession();
+    const params = new URLSearchParams({ context }).toString();
+    const shouldFetch = context && orderId !== undefined;
+
+    const { order, error: orderError } = useOrder(orderId);
+    // Shipping addresses and products are not included in the order data and need to be fetched separately
+    const { data: shippingAddresses, error: shippingError } = useSWR<ShippingAddress[], ErrorProps>(
+        shouldFetch ? [`/api/orders/${orderId}/shipping_addresses`, params] : null, fetcher
+    );
+    const { data: products, error: productsError } = useSWR<OrderProduct[], ErrorProps>(
```

Comment:
imho, there's a lot going on with this hook. It seems like we're trying to cram all of the logic here, rather than having multiple, smaller functions that do just one thing. This can create issues with readability and maintainability. Moreover, if we're trying to package multiple API calls into one object, why not do that in the API level (via Promise.all)? That would seem like a more appropriate place to bundle the `shippingAddress` and `products` call, rather than creating multiple useSWR hooks here. Doing so in the API level will also allow us to simplify our `isLoading`, and `error` objects below.